Sources/AWSSDKSwiftCore/XML.swift

```diff
@@ -0,0 +1,465 @@
+        public static func namespace(withName: String? = nil, stringValue: String) -> XML.Node {
+            return XML.Node(.namespace, name: withName, stringValue: stringValue)
+        }
+        
+        /// create XML comment node
+        public static func comment(stringValue: String) -> XML.Node {
+            return XML.Node(.comment, stringValue: stringValue)
+        }
+        
+        /// return child node at index
+        private func child(at index: Int) -> XML.Node? {
+            return children?[index]
+        }
+        
+        /// return number of children
+        public var childCount : Int { get {return children?.count ?? 0}}
+        
+        /// detach XML node from its parent
+        public func detach() {
+            parent?.detach(child:self)
+            parent = nil
+        }
+        
+        /// detach child XML Node
+        fileprivate func detach(child: XML.Node) {
+            children?.removeAll(where: {$0 === child})
+        }
+        
+        /// return children of a specific kind
+        func children(of kind: Kind) -> [XML.Node]? {
+            return children?.compactMap { $0.kind == kind ? $0 : nil }
+        }
+        
+        private static let xmlEncodedCharacters : [String.Element: String] = [
+            "\"": "&quot;",
+            "&": "&amp;",
+            "'": "&apos;",
+            "<": "&lt;",
+            ">": "&gt;",
+        ]
+        /// encode text with XML markup
+        private static func xmlEncode(string: String) -> String {
+            var newString = ""
+            for c in string {
+                if let replacement = XML.Node.xmlEncodedCharacters[c] {
+                    newString.append(contentsOf:replacement)
+                } else {
+                    newString.append(c)
+                }
+            }
+            return newString
+        }
+        
+        /// output formatted XML
+        public var xmlString : String {
+            switch kind {
+            case .text:
+                if let stringValue = stringValue {
+                    return XML.Node.xmlEncode(string: stringValue)
+                }
+                return ""
+            case .attribute:
+                if let name = name {
+                    return "\(name)=\"\(stringValue ?? "")\""
+                } else {
+                    return ""
+                }
+            case .comment:
+                if let stringValue = stringValue {
+                    return "<!--\(stringValue)-->"
+                } else {
+                    return ""
+                }
+            case .namespace:
+                var string = "xmlns"
+                if let name = name, name != "" {
+                    string += ":\(name)"
+                }
+                string += "=\"\(stringValue ?? "")\""
+                return string
+            default:
+                return ""
+            }
+        }
+        
+        /// CustomStringConvertible protocol
```

Comment:
You could move these into extensions if you'd like, that would help convey this meaning via code, which is nice.

```swift
extension XML: CustomStringConvertible {
```

Reply:
Done